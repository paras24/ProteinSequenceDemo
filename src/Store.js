import {createStore} from 'redux'
import rootReducers from './redux/rootReducer'
import { applyMiddleware } from 'redux'
import {logger} from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunk from 'redux-thunk'

// Added redux-logger for logging purpose, we can check logging inside browser console
// Added redux-thunk for using creating Async Actions
// Added redux-devtools-extension for easily debuging of State change and Action calls inside browser developer tools

export const store = createStore(
    rootReducers,
     composeWithDevTools(applyMiddleware(logger,thunk))
     )
