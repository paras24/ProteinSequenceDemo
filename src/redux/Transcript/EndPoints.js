// We can add all endpoints in this file

export const getTranscritUrl= (symbolName)=> `http://rest.ensembl.org/lookup/symbol/homsap/${symbolName}.json?;expand=1`

