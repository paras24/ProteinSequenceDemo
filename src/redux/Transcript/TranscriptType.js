const FETCH_TRANSCRIPT_REQ = 'FETCH_TRANSCRIPT_REQ'
const FETCH_TRANSCRIPT_SUCCESS = 'FETCH_TRANSCRIPT_SUCCESS'
const FETCH_TRANSCRIPT_ERROR = 'FETCH_TRANSCRIPT_ERROR'

// Created types for various actions and export it for further usage
export {FETCH_TRANSCRIPT_REQ,FETCH_TRANSCRIPT_SUCCESS,FETCH_TRANSCRIPT_ERROR}