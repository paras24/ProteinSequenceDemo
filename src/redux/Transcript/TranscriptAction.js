import {FETCH_TRANSCRIPT_REQ,
        FETCH_TRANSCRIPT_SUCCESS,
         FETCH_TRANSCRIPT_ERROR}
             from './TranscriptType'
import axios from 'axios'
import {getTranscriptUrl} from './EndPoints'


// created various actions in order to facilate various transitions
export const fetchTransReq = ()=>{
return{
    type:FETCH_TRANSCRIPT_REQ
}
}

export const fetchTransSuccess = (transcripts)=>{
return{
    type:FETCH_TRANSCRIPT_SUCCESS,
    Transcripts : transcripts
}}

export const fetchTransError = (err) =>{
return {
    type: FETCH_TRANSCRIPT_ERROR,
    error: err
}}


// Created async action & dispatching various action requests
// Using axios in order to deal with various Api's call
export const fetchTranscripts = (symbol)=>{
    const symbolName = symbol
    const transcriptUrl = getTranscriptUrl(symbolName)
    return (dispatch)=>{
    dispatch(fetchTransReq())
    axios.get(transcriptUrl)
    .then(res => 
        {
            const transcripts = res.data            
            dispatch(fetchTransSuccess(transcripts))
        })
    .catch(error =>{
        const errorMessage = error.response.data.error
        dispatch(fetchTransError(errorMessage))
    })
}}