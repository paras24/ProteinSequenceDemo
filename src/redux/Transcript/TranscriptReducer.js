import {FETCH_TRANSCRIPT_REQ,FETCH_TRANSCRIPT_SUCCESS,FETCH_TRANSCRIPT_ERROR} from './TranscriptType'

const intialState = {
    loading:false,
    transcripts:{},
    error:''
}

const transcriptReducer = (state = intialState , action)=>{
   switch(action.type){
       case FETCH_TRANSCRIPT_REQ : return{
           ...state,
           loading :true
       }
       case FETCH_TRANSCRIPT_SUCCESS : return{
           loading :false,
           transcripts : action.Transcripts,
           error : ''
       }
       case FETCH_TRANSCRIPT_ERROR : return {
           loading : false,
           transcripts : {},
           error : action.error
       }
       default : return state
   }
}


export default transcriptReducer