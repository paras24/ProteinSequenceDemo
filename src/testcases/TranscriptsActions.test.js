import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import {fetchTransSuccess,
          fetchTransReq,
           fetchTransError,
              fetchTranscripts} from '../redux/Transcript/TranscriptAction'
import {store} from '../Store'
import {getTranscritUrl} from '../redux/Transcript/EndPoints'

describe('fetchTransSuccess', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(fetchTransSuccess('BRAF'));
      expect(store.dispatch(fetchTransSuccess('BRAF'))).toMatchSnapshot();
    });
})

describe('fetchTransReq', () => {
  test('Dispatches the correct action and payload', () => {
    store.dispatch(fetchTransReq());
    expect(store.dispatch(fetchTransReq())).toMatchSnapshot();
  });
})

describe('fetchTransError', () => {
  test('Dispatches the correct action and payload', () => {
    store.dispatch(fetchTransError('Something Went Wrong'));
    expect(store.dispatch(fetchTransError('Something Went Wrong'))).toMatchSnapshot();
  });
})


describe('fetchTranscriptsWithSymbol', () => {
it('works with async actions', async () => {
    const data = await fetchTranscripts('BRAF');
    expect(data.length > 0);
  });
})


describe('fetchTranscriptsWithoutSymbol', () => {
  it('works with async actions', async () => {
      const data = await fetchTranscripts(' ');
      expect(data.length == 0);
    });
  })


describe('Get Transcripts', () => {
    it('Verify transcript URL', async () => {
      const symbolName = 'BRAF'
      const result = getTranscritUrl(symbolName)  
      expect(result.endsWith(`/${symbolName}.json?;expand=1`))
     });
  })
  
