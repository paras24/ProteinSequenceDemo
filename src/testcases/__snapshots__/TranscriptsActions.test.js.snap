// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch Transcript Error Dispatches the correct action and payload 1`] = `
Object {
  "error": "Something Went Wrong",
  "type": "FETCH_TRANSCRIPT_ERROR",
}
`;

exports[`fetchTransError Dispatches the correct action and payload 1`] = `
Object {
  "error": "Something Went Wrong",
  "type": "FETCH_TRANSCRIPT_ERROR",
}
`;

exports[`fetchTransReq Dispatches the correct action and payload 1`] = `
Object {
  "type": "FETCH_TRANSCRIPT_REQ",
}
`;

exports[`fetchTransSuccess Dispatches the correct action and payload 1`] = `
Object {
  "Transcripts": "BRAF",
  "type": "FETCH_TRANSCRIPT_SUCCESS",
}
`;
